[tensorexp] Add strides to Buf

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68313
* __->__ #68018

Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

IvanKobzarev committed:
[tensorexp] Add strides to Buf

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

ZolotukhinM commented on torch/csrc/jit/tensorexpr/ir.cpp:
```diff
@@ -83,7 +83,8 @@ StorePtr BufHandle::store(
 ExprPtr flatten_index(
     const std::vector<ExprPtr>& dims,
-    const std::vector<ExprPtr>& indices) {
+    const std::vector<ExprPtr>& indices,
+    const std::vector<ExprPtr>& strides) {
```
We don't need `dims` anymore I think?

IvanKobzarev replied:
Yes, if strides are not optional here - we can remove dims.

ZolotukhinM reviewed (approved):
This looks good to me! I think we should take another look later on at how strides are modeled/used in our codebase, but this implementation doesn't seem to interfere with any existing component.

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

t10-13rocket reviewed (approved):
LGTM

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).

IvanKobzarev committed:
Update on "[tensorexp] Add strides to Buf"


Differential Revision: [D32262381](https://our.internmc.facebook.com/intern/diff/D32262381)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32262381).
